lib/ansible/modules/cloud/ovh/ovh_api.py

```diff
@@ -0,0 +1,132 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = '''
+---
+module: ovh_api
+version_added: "2.10"
+author: Francois Lallart (@fraff)
+short_description: Minimalist wrapper around OVH api and ovh python module
+description:
+    - Minimalist wrapper around OVH api and ovh python module
+requirements: [ "ovh" ]
+options:
+    path:
+        required: true
+        type: str
+        description:
+            - https://api.ovh.com/console/{path} don't forget to replace / with %2f
+    method:
+        required: false
+        type: str
+        choices: ['GET', 'POST', 'PUT', 'DELETE']
+        default: GET
+        aliases: ['action']
+        description:
+            - obvious
+    body:
+        required: false
+        type: dict
+        default:
+        aliases: ['data']
+        description:
+            - use this body if required
+'''
+
+EXAMPLES = '''
+# list your cloud projects
+- ovh_api: path="/cloud/project"
+
+# set a reverse to your x.x.x.x/29 ip block
+- ovh_api:
+    path: "/ip/x.x.x.x%2f29/reverse"
+    method: "POST"
+    body: {'ipReverse': 'y.y.y.y', reverse: 'my.cool.reverse'}
+
+# get info about yourself
+- ovh_api: method=GET path=/me
+'''
+
+RETURN = '''
+'''
+
+import os
+import sys
+
+try:
+    import ovh
+    import ovh.exceptions
+    from ovh.exceptions import APIError
+    HAS_OVH = True
+except ImportError:
+    HAS_OVH = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(required=True),
+            method=dict(required=False, default='GET', choices=['GET', 'POST', 'PUT', 'DELETE'], aliases=['action']),
+            body=dict(required=False, default={}, type='dict', aliases=['data'])
+        ),
+        supports_check_mode=False
+    )
+
+    if not HAS_OVH:
+        module.fail_json(msg='python-ovh is required to run this module, see https://github.com/ovh/python-ovh')
+
+    # Get parameters
+    path = module.params.get('path')
+    method = module.params.get('method').lower()
+    body = module.params.get('body')
+
+    # Connect to OVH API
+    client = ovh.Client()
+
+    mydict = {
+        'func': {'get': client.get, 'put': client.put, 'post': client.post, 'delete': client.delete},
+        'code': {'get': False, 'put': True, 'post': True, 'delete': True}
+    }
+
+    try:
+        result = mydict['func'][method](path, **body)
+        module.exit_json(changed=mydict['code'][method], result=result)
```

Comment:
I was sure ovh python module returns true if something is modified and error if nothing modified (because non existing object or something) but I've just found a PUT that always returns 'null', I'll ask the OVH guys to clarify